### PR TITLE
refactor(scripts/template/initialize): migrate to TypeScript

### PR DESCRIPTION
**Pull request by hoxyq**
Summary: Changelog: [Internal]

Differential Revision: D42781211

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,10 +97,13 @@
     "test-e2e-local": "node ./scripts/test-e2e-local.js",
     "test-e2e-local-clean": "node ./scripts/test-e2e-local-clean.js",
```
Comment (kelset) [inline package.json:98]: headsup: we should migrate to TS also the other scripts in the folder like these two above 👆 is the plan to do it all at once or partially across multiple PRs?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1490,7 +1495,12 @@ jobs:
                 name: Build release package as a job artifact
                 command: |
                   mkdir -p build
-                  FILENAME=$(npm pack)
+
+                  # Run prepack script manually, instead of providing --unsafe-perm option to npm
+                  yarn prepack
+
+                  FILENAME=$(npm pack | tail -1)
```
Comment (kelset) [inline .circleci/config.yml:1502]: this is technically unrelated to these changes right? is it a different minor improvement that you want to bring in? or there's some direct reason?

**Review by kelset — COMMENTED** [review]
overall I'm 👍 with these changes, let's just make sure to plan accordingly to cover all the *.js files (if all in this PR or if in multiples) in the `/scripts` folder - and that the TS config is aligned with the rest of react-native, and the template. Meaning, basically, let's make sure that @NickGerleman is 👍 too with the config

File: package.json (hunk 1 above)
Comment (hoxyq) [reply package.json:98]: The plan for now is to migrate partially across multiple PRs, just for me to deliver some constant increment on this and not being fully invested in this JS -> TS migration. Is there any requirements on migrating all at once or one after another?

There is only one script I know for now which we won't be able to migrate, its `monorepo/find-and-publish-all-bumped-packages.js`, because it need to be executed without running `yarn`, so no `ts-node` will be present.

Comment (kelset) [reply package.json:98]: ok sounds good - as long as we keep track of it all then I'm 👍 I think we can proceed as soon as we are sure that it's also ✅ from Meta side :)

**Review by kelset — APPROVED** [review]
:shipit:
